File: Python3.x/14-Longest_Common_Prefix.py
# Runtime: 32 ms, faster than 98.04% of Python3 online submissions for Longest Common Prefix.
# Memory Usage: 13.2 MB, less than 67.21% of Python3 online submissions for Longest Common Prefix.
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 0 or len(strs[0]) == 0: return ""
        min_len = min(len(s) for s in strs)
        res = ""
        cont = True
        for i in range(min_len):
            tmp = strs[0][i]
            same = True
            for s in strs[1:]:
                if s[i] != tmp:
                    same, cont = False, False
                    break
            if same and cont:
                res += tmp
        return res

File: Algorithms/C++/14-Longest_Common_Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string prefix;
        if (strs.size() == 0)
            return prefix;
        bool is_same = true;
        for (int i = 0; i < strs[0].size() && is_same == true; ++i) {
            for (int j = 1; j < strs.size(); ++j)
                if (strs[j - 1][i] != strs[j][i]) {
                    is_same = false;
                    break;
                }
            if (!is_same)
                break;
            prefix += strs[0][i];
        }
        return prefix;
    }
};
